### PR TITLE
Issue617 refactor service

### PR DESCRIPTION
**Pull request by dhblum**
This is a proposal for #617.  So far, it:

- migrates [boptest-service](https://github.com/NREL/boptest-service) into its own directory within the repo
- removes ``api.py``
- sets up the ``docker-compose.yml`` and other Dockerfiles to work with the new directly structure.  
- cites the boptest-service repo license
- updates unit test and example interfaces (except javascript) to pass.

See #617 for remaining todos, especially documentation/readme updates.

@kbenne @javiarrobas Care to provide initial review?

### CODE REVIEW

File: service/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# BOPTEST-Service
+
+This software extends [BOPTEST](https://github.com/ibpsa/project1-boptest) to a web service architecture, which enables support for multiple clients and multiple simultaneous tests at a large scale. This is a containerized design that can be deployed on a personal computer, however the software is targeted at commercial cloud computing environments such as AWS. For details about BOPTEST, refer to the project [homepage](https://boptest.net).
+
+The canonical BOPTEST source code is incorporated into this repository as a git subtree located at `<project-root>/boptest` and used under the terms of the license located at `<project-root>/boptest/license.md`.
```
Comment (javiarrobas) [inline service/README.md:5]:
```suggestion
This service uses the canonical BOPTEST source code located at the root of this repository under the terms of the license located at `<project-root>/boptest/license.md.
```

File: service/worker/__main__.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+print("Starting Alfalfa Worker")
```
Comment (javiarrobas) [inline service/worker/__main__.py:1]:
```suggestion
print("Starting BOPTEST Worker")
```

**Review by javiarrobas — APPROVED** [review]
Looking good to me @dhblum! just added a couple of small suggestions.

**Comment by dhblum** [comment]
Thank you @javiarrobas!  I'll take a look at your comments/suggestions, also with @kbenne, who plans to also see if he can replicate the migration but in a way to maintain the Service git history (in this PR I just simply moved files from that repo into this one).

File: .env [hunk 3]
```diff
@@ -1,3 +1,34 @@
...
+NODE_ENV=production
+BOPTEST_TIMEOUT=900
+BOPTEST_LOGLEVEL=INFO
+
+# Redis configuration
+BOPTEST_REDIS_HOST=redis
+
+# Job queue configuration
+BOPTEST_JOB_QUEUE_URL=http://goaws:4100/queue/local-queue1
+AWS_ACCESS_KEY_ID=xxxxxxxxxxxxxxxxxxxx
```
[the remark this answers is not in the document]
Comment (dhblum) [reply .env:11]: @kbenne Thoughts?

File: service/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+| Select a test case from the namespace and begin a new test. (Auth optional)                                                 | POST ``testcases/{namespace}/{testcase_name}/select``      |
+| Select a private user test case and begin a new test. (Auth required)                                                       | POST ``users/{username}/testcases/{testcase_name}/select`` |
+| Get test status as `Running` or `Queued`                                                                                    | GET ``status/{testid}``                                    |
+| Stop a queued or running test.                                                                                              | PUT ``stop/{testid}``                                      |
+| List tests for a user. (Auth required)                                                                                      | GET ``users/{username}/tests``                             |
+
+The family of the `select` APIs are used to choose a test case and begin a running test. Select returns a `testid` which is required by all APIs that interact with the test or provide test information.
+
+# OpenAI Gym Interface
```
[the remark this answers is not in the document]
Comment (dhblum) [reply service/README.md]: Done in https://github.com/ibpsa/project1-boptest/pull/622/commits/bdb6322ef5cc22624a7ba676f07bb566b018aefc.

**Comment by dhblum** [comment]
Closing, continuing in https://github.com/ibpsa/project1-boptest/pull/652.

**Pull request closed without merging**